gprofiler/merge.py

```diff
@@ -253,15 +267,30 @@ def concatenate_profiles(
     """
     total_samples = 0
     lines = []
+    application_metadata: List[Optional[Dict]] = [None]
 
     for pid, stacks in process_profiles.items():
-        container_name = _get_container_name(pid, container_names_client, add_container_names)
-        application_name = get_application_name(pid) if identify_applications else None
+        # generate application name
+        application_name = get_application_name(pid) if enrichment_options.application_identifiers else None
         if application_name is not None:
             application_name = f"appid: {application_name}"
-        prefix = (container_name + ";") if add_container_names else ""
```

Comment:
Previously - we'd always add `;` if protocol version required it; Now we don't. Same for app metadata. I'll clarify that with the backend

Reply:
I will keep that, and also keep protocol version v2 (i.e revert all changes relating to protocol version) as the backend will decide on existence of this field based on the `application_metadata: true` existence.

Reply:
Discussed in https://github.com/Granulate/gprofiler/pull/299#issuecomment-1063493241